tesseract/tesseract_collision/src/fcl/fcl_discrete_managers.cpp

```diff
@@ -155,21 +154,98 @@ void FCLDiscreteBVHManager::setCollisionObjectsTransform(const std::string& name
 {
   auto it = link2cow_.find(name);
   if (it != link2cow_.end())
-    it->second->setCollisionObjectsTransform(pose);
+  {
+    const Eigen::Isometry3d& cur_tf = it->second->getCollisionObjectsTransform();
+    // Note: If the transform has not changed do not updated to prevent unnecessary rebalancing of the BVH tree
+    if (!cur_tf.translation().isApprox(pose.translation(), 1e-8) || !cur_tf.rotation().isApprox(pose.rotation(), 1e-8))
+    {
+      it->second->setCollisionObjectsTransform(pose);
+      if (it->second->m_collisionFilterGroup == CollisionFilterGroups::StaticFilter)
+      {
+        // Note: Calling update causes a rebalance of the AABB tree, which is expensive
+        static_manager_->update(it->second->getCollisionObjectsRaw());
+      }
+      else
+      {
+        // Note: Calling update causes a rebalance of the AABB tree, which is expensive
+        dynamic_manager_->update(it->second->getCollisionObjectsRaw());
+      }
+    }
+  }
```

Comment:
While digging through the bullet code I noticed that they were checking the transform to make sure it actually moved to prevent unnecessary re-balance of the bvh. I an not sure if this would be a performance hit.